solidity/contracts/hooks/ArbitrumOrbitHook.sol

```diff
@@ -0,0 +1,130 @@
+/// The terms L1 and L2 used here to mean the source and destination chain, so L1->L2 can refer to L1->L2, L2->L3, L3->L4 and so on.
+/// You can find an introduction to Hyperlane's message passing at https://docs.hyperlane.xyz/docs/reference/messaging/messaging-interface.
+/// @notice When the Mailbox calls this object, the mechanism to realize L1->L2 message passing is through Arbitrum's retryables.
+/// At a high level, when Mailbox calls this object, this object creates a retryable,
+/// Then Arbitrum **strongly gurantees the retryable will eventually be executed successfully**.
+/// You can find retryable docs and its lifecycle at https://docs.arbitrum.io/arbos/l1-to-l2-messaging#submission.
+/// @dev This object is meant to be called as custom hook. So as an end user, you will want to
+/// give this object's address as custom hook when calling the Mailbox.
+/// Docs: https://docs.hyperlane.xyz/docs/reference/hooks/overview#custom-hook-and-metadata.
+contract ArbitrumOrbitHook is AbstractMessageIdAuthHook {
+    using Message for bytes;
+    using StandardHookMetadata for bytes;
+
+    IInbox public immutable l1Inbox;
+    IGasOracle public immutable gasOracle;
+
+    /// @dev See StandardHookMetadata.sol
+    uint256 private constant CUSTOM_METADATA_OFFSET = 86;
+
+    event RetryableTicketCreated(uint256 indexed ticketId);
+
+    constructor(
+        address _mailbox,
+        uint32 _destinationDomain,
+        bytes32 _ism,
+        address _l1Inbox,
+        address _gasOracle
+    ) AbstractMessageIdAuthHook(_mailbox, _destinationDomain, _ism) {
+        require(Address.isContract(_l1Inbox), "ArbitrumHook: invalid inbox");
+        l1Inbox = IInbox(_l1Inbox);
+        gasOracle = IGasOracle(_gasOracle);
+    }
+
+    /**
+     * @notice Send a message to the ISM.
+     * @dev This is ultimately called from postDispatch in IPostDispatchHook.
+     * @param metadata Docs: https://docs.hyperlane.xyz/docs/reference/hooks/overview#custom-hook-and-metadata
+     * metadata format:
+     * [0:2] variant (meaning the 0th to 2nd byte is the variant)
+     * [2:34] msg.value
+     * [34:66] Gas limit for message
+     * [66:86] Refund address for message
+     * [86:117] Optional max fee per gas (this is custom metadata)
+     * @param payload The payload for call to the ISM.
+     */
+    function _sendMessageId(
+        bytes calldata metadata,
+        bytes memory payload
+    ) internal override {
+        address refundAddress = metadata.refundAddress(address(0));
+        uint256 ticketID = l1Inbox.createRetryableTicket{value: msg.value}(
+            TypeCasts.bytes32ToAddress(ism),
+            metadata.msgValue(0),
+            _submissionFee(),
+            refundAddress,
+            refundAddress,
+            metadata.gasLimit(0),
+            _maxFeePerGas(metadata),
+            payload
+        );
+        emit RetryableTicketCreated(ticketID);
```

Comment:
does the arbitrum inbox emit an event already? this may not be necessary

Reply:
removed